port_definition.go

```diff
@@ -0,0 +1,24 @@
+/*
+Copyright 2014 Rohith All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package marathon
+
+// PortDefinition provides metadata about a port
+type PortDefinition struct {
+	Port     int    `json:"port"`
```

Comment:
I think we need to turn this into a pointer in order to distinguish between omitting, zero, and non-zero as the three carry different semantics (to my understanding) and Marathon only does PATCH-like updates even though everything is PUT. (For an extended rationale, see [this blog post on Go and REST APIs](https://willnorris.com/2014/05/go-rest-apis-and-pointers).)

Would you mind making the change and adding a convenience setter method on `PortDefinition` similar to [Application.Mem()](https://github.com/gambol99/go-marathon/blob/master/application.go#L175) (plus a quick test)?